core/src/types.ts

```diff
@@ -266,3 +266,40 @@ export declare interface DebtAuctionEnriched extends DebtAuctionActive {
 }
 
 export declare interface DebtAuctionTransaction extends DebtAuctionEnriched, CompensationAuctionTransactionFees {}
+
+type CollateralType = CollateralConfig['title'];
+export declare interface Collateral {
+    type: CollateralType;
+    stabilityFeeRate: BigNumber;
+    maxDebtPerCollateralUnit: BigNumber;
+    liquidationLimits: LiquidationLimit;
+}
+export declare interface VaultAddress {
+    id: number;
+    address: string;
+    collateralType: CollateralType;
+}
+export declare interface VaultDebtAmount {
+    initialDebt: BigNumber;
```

Comment:
debt = generated dai (initial debt) + stability fees.

Since this type represents the information that is initially fetched from the blockchain, `debt` should not be put here because contract does not store this information. Instead it stores the stability fees' rate and the initially extracted dai. These two can be used to compute the overall debt.

Reply:
> These two can be used to compute the overall debt.

Should we then add both `initialDebtDai` and `debtDai`?

Reply:
well, we can, but not in this type because then it would require the rates fetched beforehand. Added this to the `Vault`